### PR TITLE
[LI-HOTFIX] Use version 1 fields of LiCombinedControl request.

### PR DESCRIPTION
**Pull request by wyuka**
When running Kafka version >= 3.0, use the version 1 (new) of `LiCombinedControlRequest`. This has the following changes -
1. Send `leaderEpoch` as part of `StopReplicaPartitionState` only if using. version >= 1.
2. Send `StopReplicaTopicStates` instead of `StopReplicaPartitionStates` only if using version >= 1.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/controller/ControllerRequestMergerTest.scala [hunk 1]
```diff
@@ -285,6 +301,44 @@ class ControllerRequestMergerTest {
     assertEquals(toMap(expectedPartitions.asJava), toMap(liCombinedControlRequest.stopReplicaPartitionStates()))
   }
 
+  @Test
+  def testMergingDifferentStopReplicaPartitionStatesV0Request(): Unit = {
```
Comment (gitlw) [inline core/src/test/scala/unit/kafka/controller/ControllerRequestMergerTest.scala]: The logic is quite similar to the testMergingDifferentStopReplicaPartitionStates method.
Can we extract the common logic into one shared method with the two parameters: 1) the IBP version and 2) shouldSetLeaderEpoch.

Comment (wyuka) [reply core/src/test/scala/unit/kafka/controller/ControllerRequestMergerTest.scala]: Thanks, that is a really good suggestion. On it!

File: core/src/test/scala/unit/kafka/controller/ControllerRequestMergerTest.scala [hunk 2]
```diff
@@ -417,3 +444,12 @@ class ControllerRequestMergerTest {
...
 }
+
+object ControllerRequestMergerTest {
+  def testMergingDifferentStopReplicaPartitionStatesParams: java.util.stream.Stream[Arguments] = {
+    Seq(
+      Arguments.of("2.4".asInstanceOf[AnyRef], false.asInstanceOf[AnyRef]),
+      Arguments.of("3.0".asInstanceOf[AnyRef], true.asInstanceOf[AnyRef]),
+    ).asJava.stream()
+  }
+}
```
Comment (lmr3796) [inline core/src/test/scala/unit/kafka/controller/ControllerRequestMergerTest.scala]: nit: should have a new-line character at EOF (I'd recommend setting this up in IDE).

**Review by lmr3796 — CHANGES_REQUESTED** [review]
A small nit.  Otherwise LGTM.

**Comment by lmr3796** [comment]
Also when squashing remember to remove the period in subject line :P

**Review by gitlw — APPROVED** [review]
LGTM after fixing the nit comment by Joseph.
Thanks for this patch!

**Pull request merged**